src/pkg/services/m365/api/pagers/pagers.go

```diff
@@ -379,37 +382,116 @@ type addedAndRemovedHandler[T any] func(
 	error,
 )
 
+func getLimitedItems[T any](
```

Comment:
the name here doesn't feel quite right, since it sounds like the items are what's limited.
```suggestion
func batchWithMaxItemCount[T any](
```

Reply:
Could you also add a function description?